cloudinit/cmd/devel/logs.py

```diff
@@ -242,7 +250,8 @@ def collect_logs(tarfile, include_userdata: bool, verbosity=0):
             verbosity=verbosity,
         )
 
-        for log in CLOUDINIT_LOGS:
+        init.read_cfg()
+        for log in get_config_logfiles(init.cfg):
```

Comment:
I do not think `get_config_logfiles` is aware of the rotated log files, so this functionality needs to be added there.

Reply:
Added logic to `get_config_logfiles` function to take care of rotated logfiles for cloud-init as well as cloud-init-output log. Rotated logfiles will only be added to list when they exist. This step is necessary to be compliant with diverse unit tests which test or mock this function. Let me know what you think :)